src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs

```diff
@@ -1,75 +1,135 @@
+        #endregion
+
+        #region Asia
+        public static readonly Region ASIA_EAST = new Region("eastasia");
+        public static readonly Region ASIA_SOUTHEAST = new Region("southeastasia");
+        public static readonly Region JAPAN_EAST = new Region("japaneast");
+        public static readonly Region JAPAN_WEST = new Region("japanwest");
+        public static readonly Region AUSTRALIA_EAST = new Region("australiaeast");
+        public static readonly Region AUSTRALIA_SOUTHEAST = new Region("australiasoutheast");
+        public static readonly Region INDIA_CENTRAL = new Region("centralindia");
+        public static readonly Region INDIA_SOUTH = new Region("southindia");
+        public static readonly Region INDIA_WEST = new Region("westindia");
+        #endregion
+
+        #region China
+        public static readonly Region CHINA_NORTH = new Region("chinanorth");
+        public static readonly Region CHINA_EAST = new Region("chinaeast");
+        #endregion
+
+        #region German
+        public static readonly Region GERMANY_CENTRAL = new Region("germanycentral");
+        public static readonly Region GERMANY_NORTHEAST = new Region("germanynortheast");
+        #endregion
+
+        #region Government Cloud
+        public static readonly Region GOV_US_VIRGINIA = new Region("usgovvirginia");
+        public static readonly Region GOV_US_IOWA = new Region("usgoviowa");
+        #endregion
+
+        public static IReadOnlyCollection<Region> Values
+        {
+            get
+            {
+                return regions.Values as IReadOnlyCollection<Region>;
+            }
+        }
+
+        public string Name
+        {
+            get; private set;
+        }
+
+        private Region(string name)
+        {
+            Name = name.ToLowerInvariant();
+            regions.Add(Name, this);
+        }
+
+        public static Region Create(string name)
+        {
+            name = sWhitespace.Replace(name, "").ToLowerInvariant();
+            Region region = null;
+            if (regions.TryGetValue(name, out region))
+            {
+                return region;
+            }
+            return new Region(name);
+        }
+
+        public override int GetHashCode()
+        {
+            return this.Name.GetHashCode();
+        }
+
+        public static bool operator ==(Region lhs, Region rhs)
+        {
+            if (object.ReferenceEquals(lhs, null))
+            {
+                return object.ReferenceEquals(rhs, null);
+            }
+            return lhs.Equals(rhs);
+        }
+
+        public static bool operator !=(Region lhs, Region rhs)
+        {
+            return !(lhs == rhs);
+        }
+
+        public override bool Equals(object obj)
+        {
+
+            string value = this.ToString();
+            if (!(obj is Region))
+            {
+                return false;
+            }
+
+            if (obj == this)
```

Comment:
>if (obj == this) [](start = 12, length = 16)

this comparison produces a valid warning "Possible unintended reference comparison; to get a value comparison, cast the left hand side to type". Please fix it here and in all other cases that you've already merged.